src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignContigsAndCallBreakpointsSpark.java

```diff
@@ -0,0 +1,121 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.bwa.BwaSparkEngine;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import scala.Tuple2;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+@CommandLineProgramProperties(summary="Align assembled contigs to the reference and call breakpoints from them.",
+        oneLineSummary="Align contigs and call breakpoints",
+        programGroup = SparkProgramGroup.class)
+public class AlignContigsAndCallBreakpointsSpark extends GATKSparkTool {
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Number of assemblies to process per task. We need to make sure that the work is not
+     * over-partitioned since each worker needs to localize the BWA reference index and read it into
+     * memory once per partition.
+     */
+    public static final int NUM_ASSEMBLIES_PER_PARTITION = 200;
+
```

Comment:
Not a comment against the code or design here, but I was considering edge cases when this could fail. One such scenario is that when SPARK is run in local mode, and there are 1,000 contigs FASTA file to be processed -> 5 partitions. Each one, loading the reference, could blowup the memory usage?